### ISSUE TITLE
Bug: Missing build step in `contributing.md`? Not sure what it is

### ISSUE THREAD

**Issue by ayjayt** [plotly/plotly.py#4518]
edit: Sorry yeah build instructions from `contributing.md` don't seem to work.

i installed `requirements.txt` and `optional-requirements.txt` and then ran

`pip install -e packages/python/plolty`

```bash

> webpack --mode=production

rebuilding js and css failed
missing files: ['/home/ajp/projects-deps/plotly.py/packages/python/plotly/jupyterlab_plotly/nbextension/index.js',
 '/home/ajp/projects-deps/plotly.py/packages/python/plotly/jupyterlab_plotly/labextension/package.json']
```

Thanks!

**Comment by alexcjohnson** [comment]
Thanks @ayjayt.

@LiamConnors can you tell what steps we're missing?

**Comment by LiamConnors** [comment]
@ayjayt  I went through these steps yesterday in a Conda environment running Python 3.10 and didn't encounter the above error. Is the error from when you run the `python setup.py updateplotlyjs`command?

**Comment by ayjayt** [comment]
I'll run through it again hopefully this afternoon and I'll also take a
look at the circleci build steps for clues

**Comment by Coding-with-Adam** [comment]
Thanks for taking another look at it, @ayjayt . If this keeps happening let me know and could try to solve this on a quick call. adam@plot.ly

**Comment by ayjayt** [comment]
## ISSUE 1: `pip install -e ./packages/python/plotly`
Okay, the issue is that sometimes `webpack --mode=production` for `npm run build:nbextension` just fails silently, causing a cascade of failures. I think it's a matter of resource consumption: when i split the build into one `entrypoint` at a time, it works. `setup.py` in `python/plotly` calls `npm run build:nbextension` but doesn't raise if it fails.

 may alter `setup.py`to raise if one of it's dependencies fails, as it should! would explain why builds work for you, bigger machines.

## ISSUE 2: `pip install -r ./packages/python/plotly/optional-requirements.txt`

ALSO, @LiamConnors, youre using 3.10. and yeah, the new version of python (3.12)  will fail during `pip install -r bla/bla/bla/optional-requirements.txt` because of:
- `jupyterlab ~3.0`  (which resolves today to `3.6.7`) due to dependency `pyzmq`. related issue: https://github.com/cython/cython/issues/5238

and
- `matplotlib 2.2.2` due to dependency `versioneer`  (commented out anyway, but probs necessary for `matplotlylib`) related issue: https://github.com/pydata/pandas-datareader/issues/969#issuecomment-1786843855

**Comment by ayjayt** [comment]
Since it's a resource issue and not really something you all have control over, I'm going to close this issue. May be interested in my other notes if 3.12 support doesn't improve over time.
